### PR TITLE
[SPARK-36692][CORE] Improve Error statement when requesting thread dump while executor already stopped

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
For now, when user check thread dump for a executor while this executor is stopped, the error log shows following might disturb users.

image

### Why are the changes needed?
Improve error message

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @HyukjinKwon

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -694,8 +694,14 @@ class SparkContext(config: SparkConf) extends Logging {
...
         Some(Utils.getThreadDump())
       } else {
-        val endpointRef = env.blockManager.master.getExecutorEndpointRef(executorId).get
-        Some(endpointRef.askSync[Array[ThreadStackTrace]](TriggerThreadDump))
+        env.blockManager.master.getExecutorEndpointRef(executorId) match {
+          case Some(endpointRef) =>
+            Some(endpointRef.askSync[Array[ThreadStackTrace]](TriggerThreadDump))
+          case None =>
+            logWarning(s"Executor $executorId might already have stopped and" +
+              s" can not request thread dump from it.")
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/SparkContext.scala:702]:
```suggestion
              " can not request thread dump from it.")
```

**Review by HyukjinKwon — APPROVED** [review]
looks good. cc @mridulm @Ngone51 @tgravescs FYI

**Review by tgravescs — APPROVED** [review]

**Review by Ngone51 — APPROVED** [review]
LGTM

**Review by mridulm — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**